File: output/gemma4-e2b/avatar/Java/Python/atcoder_AGC038_A.py
import sys

def solve():
    # Read H, W, A, B from standard input
    try:
        # Read all inputs, assuming they are provided on separate lines or space-separated
        # Since the Java code uses nextInt() sequentially, we read them one by one.
        input_data = sys.stdin.read().split()
        if len(input_data) < 4:
            return
        
        H = int(input_data[0])
        W = int(input_data[1])
        A = int(input_data[2])
        B = int(input_data[3])
    except Exception:
        # Handle case where input reading fails
        return

    # Check the initial condition
    if W < 2 * A or H < 2 * B:
        print(-1)
        return

    # Generate and print the grid
    for i in range(H):
        row = []
        for j in range(W):
            # Condition for '0': (i < B and j < A) OR (i >= B and j >= A)
            is_zero = (i < B and j < A) or (i >= B and j >= A)
            
            if is_zero:
                row.append('0')
            else:
                row.append('1')
        
        print("".join(row))

solve()

File: output/gemma4-e2b/avatar/Python/Java/atcoder_AGC038_A.java
import java.util.Scanner;

public class atcoder_AGC038_A {

    public static void solve(int H, int W, int A, int B) {
        // Initialize the answer grid. Since the Python code initializes with "0"s, we use a 2D array of Strings.
        String[][] answer = new String[H][W];

        for (int i = 0; i < H; i++) {
            for (int j = 0; j < W; j++) {
                // Initialize based on the pattern derived from the Python code's initialization
                // The Python code initializes with "0"s implicitly by list comprehension, 
                // and then sets values based on the logic.
                // We will set the values based on the derived logic.
                
                if (i < B) {
                    if (j < A) {
                        answer[i][j] = "0";
                    } else {
                        answer[i][j] = "1";
                    }
                } else { // i >= B
                    if (j < A) {
                        answer[i][j] = "1";
                    } else {
                        answer[i][j] = "0";
                    }
                }
            }
        }

        // Print the result, concatenating elements of each row (mimicking ''.join(answer[i]))
        for (int i = 0; i < H; i++) {
            StringBuilder row = new StringBuilder();
            for (int j = 0; j < W; j++) {
                row.append(answer[i][j]);
            }
            System.out.println(row.toString());
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Read H, W, A, B sequentially, mimicking the token reading from sys.stdin
        if (scanner.